controllers/front/listing/CategoryController.php

```diff
@@ -155,7 +158,7 @@ public function initContent()
      */
     public function getLayout()
     {
-        if (!$this->category->checkAccess($this->context->customer->id)) {
+        if (!$this->category->checkAccess($this->context->customer->id) || $this->notFound) {
```

Comment:
hmm, why not redirect to the 404 page instead?

Reply:
I guess it's bad for SEO: #21258

Reply:
Yeah, SEO, 3 devs, 3 opinions, probably 302 to 404 page would be enough, the best solution would be to have the option to select what to do in this kind of situation, same as on product page, maybe I'll create an issue about it for the future

Reply:
You shouldn't redirect to a 404 but the page itself has to return the HTTP 404 Not found Otherwise this means.
- Trying to reach /blabla
- Page found, goes to 302 with a redirection
- Woot it's a 404.

The /blabla is still found.

But maybe the right solution to add a custom `renderNotFound` in the FrontController.php, it will helps for rendering 404 page outside the CategoryController?

Reply:
@kpodemski I have zero knowledge of SEO 😄  but I think the 404 makes sense.

If I search for category "Kid clothes"
- if it exists, server tells me a HTTP 200 => it exists!
- if it does not exists, server tells me a HTTP 404 => nope, there is no such thing
- if it existed but it was moved I expect a 301 or 302

So I find it logical, following HTTP response codes logic, to return the 404. For SEO I have no idea 😅